### PR TITLE
seexpr: 2.11 -> 3.0.1

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#144627
the new version supports python3, so it is exchanged

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Fails on aarch64-linux:

```
 [ 12%] Building CXX object src/SeExpr2/CMakeFiles/SeExpr2.dir/Context.cpp.o
  g++: error: unrecognized command line option '-msse4.1'
```

but probably not too hard to fix.

File: pkgs/development/compilers/seexpr/default.nix [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+    sha256 = "sha256-r6mgyb/FGz4KYZOgLDgmIqjO+PSmneD3KUWjymZXtEk=";
   };
 
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ libGLU libpng zlib qt4 python2Packages.pyqt4 bison flex ];
+  buildInputs = [ libGLU libpng zlib qt4 python3Packages.pyqt4 python3Packages.boost bison flex ];
+
   meta = with lib; {
     description = "Embeddable expression evaluation engine from Disney Animation";
     homepage = "https://www.disneyanimation.com/technology/seexpr.html";
```
Comment (risicle) [inline pkgs/development/compilers/seexpr/default.nix:30]: This 404s. Could do with updating.

**Comment by risicle** [comment]
```
cmakeFlags = [ "-DENABLE_SSE4=OFF" ];
```

Fixes things on aarch64. TBH you may as well set it universally, because we can't guarantee the target system will have sse4 (I think our compiler wrappers might strip the sse4 enablement anyway)

File: pkgs/development/compilers/seexpr/default.nix (hunk 1 above)
Comment (Stunkymonkey) [reply pkgs/development/compilers/seexpr/default.nix:30]: what about: https://wdas.github.io/SeExpr/

**Comment by Stunkymonkey** [comment]
i added your suggestions

**Review by risicle — APPROVED** [review]

**Comment by risicle** [comment]
(builds, nixos x86_64 & aarch64)

**Comment by nrdxp** [comment]
Result of `nixpkgs-review pr 146264` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>appleseed</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seexpr</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
>     * appleseed

fails already on master

**Pull request merged**